services/users.js

```diff
@@ -96,12 +96,28 @@ async function deleteUserAndFriends (secret) {
 }
 
 async function removeFriend (userId, friendId, secret) {
-  const authentication = await authenticateUser(userId, secret)
-  if (authentication) {
+  const isAuthenticated = await UserModel.authenticateUser(userId, secret)
+  if (isAuthenticated) {
     return UserModel.removeFriendship(userId, friendId)
   }
 }
 
+async function getUserIdsFromSecrets (secrets) {
+  return Promise.all(Object.keys(secrets).map(key => {
+    const value = UserModel.getUserIdBySecret(secrets[key])
+    return value
+  }))
```

Comment:
The map function can be made slightly more readable by removing the brackets and return

`Object.keys(secrets).map(key => UserModel.getUserIdBySecret(secrets[key]))`